fili-core/src/test/groovy/com/yahoo/bard/webservice/data/dimension/impl/SearchProviderSpec.groovy

```diff
@@ -526,6 +526,31 @@ abstract class SearchProviderSpec<T extends SearchProvider> extends Specificatio
         resultsPage.numResults == 3
     }
 
+    def "An empty string filter value should return all results"() {
+        setup: "Given a filter that will match every row"
+        /* Expected rows, not necessarily in this order:
+                name: "owl", description: "this is an owl"
+                name: "hawk", description: "this is a raptor"
+                name: "eagle", description: "this is a raptor"
+                name: "kumquat", description: "this is not an animal"
+        */
```

Comment:
So you want me to make them both lists, sort them like
```java
List<String> actualOutcomes = resultsPage.pageOfData.stream().map { it.getKeyValue() }.sorted().collect(Collectors.toList())
Collections.sort(expectedOutcomes);
```
then check for equality?
I prefer the Set way because there's no need to rely on sorting the results